docs/tutorial/saxpy.rst

```diff
@@ -0,0 +1,780 @@
+.. meta::
+  :description: The SAXPY tutorial on HIP
+  :keywords: AMD, ROCm, HIP, SAXPY, tutorial
+
+*******************************************************************************
+Tutorial: SAXPY - Hello, HIP
+*******************************************************************************
+
+This tutorial explains the basic concepts of the single-source
+Heterogeneous-computing Interface for Portability (HIP) programming model and
+the essential tooling around it. It also reviews some commonalities of
+heterogenous APIs in general. This topic assumes basic familiarity with the
+C/C++ compilation model and language.
+
+Prerequisites
+=============
+
+To follow this tutorial, you'll need installed drivers and a HIP compiler
+toolchain to compile your code. Because HIP for ROCm supports compiling and
+running on Linux and Windows with AMD and NVIDIA GPUs, the combination of
+install instructions is more than worth covering as part of this tutorial. For
+more information about installing HIP development packages, see
+:doc:`/install/install`.
+
+.. _hip-tutorial-saxpy-heterogeneous-programming
+
+Heterogeneous programming
+=========================
+
+*Heterogeneous programming* and *offloading APIs* are often mentioned together. Heterogeneous programming deals with devices of varying capabilities
+simultaneously. Offloading focuses on the "remote" and asynchronous aspects of
+computation. HIP encompasses both. It exposes GPGPU (general-purpose GPU)
+programming much like ordinary host-side CPU programming and lets you move data
+across various devices.
+
+When programming in HIP (and other heterogenous APIs for that matter), remember
+that target devices are built for a specific purpose. They are designed with
+different tradeoffs than traditional CPUs and therefore have very different
+performance characteristics. Even subtle changes in code might adversely affect
+execution time.
+
+Your first lines of HIP code
+============================
+
+First, let's do the "Hello, World!" of GPGPU: SAXPY. *SAXPY* is a mathematical
+acronym; a vector equation :math:`a\cdot x+y=z` where :math:`a\in\mathbb{R}` is
```

Comment:
I think you have mentioned SAXPY's full form below. Can we move it up here.

Reply:
Added.